### DISCUSSION TITLE
Use layer-specific edge Sets from different graphs

### DISCUSSION THREAD

**Discussion by fafal-abnir** [pyg-team/pytorch_geometric#9442] [Q&A] [answered]
I have two graphs, ***G<sub>1</sub> = (V, E<sub>1</sub>)*** and ***G<sub>2</sub> = (V, E<sub>2</sub>)***, that share the same set of nodes ***V*** but have different sets of edges ***E<sub>1</sub>*** and ***E<sub>2</sub>***. Is it possible to design a GNN with two layers(for example 2 GraphSage layers), where the message passing in the first layer uses the edges of ***G<sub>1</sub>***, and the message passing in the second layer uses the edges of ***G<sub>2</sub>***?

**Comment by akihironitta** [comment, marked as answer]
Yes, I believe it is by using `edge_index1` and `edge_index2` represent $E_1$ and $E_2$, respectively, and making sure node indices are consistent between the two graphs.
